core/src/main/scala/pureconfig/package.scala

```diff
@@ -290,16 +239,7 @@ package object pureconfig {
    * @param namespace       the base namespace from which the configuration should be load
    */
   def loadConfigFromFiles[Config](files: Traversable[Path], failOnReadError: Boolean = false, namespace: String = "")(implicit reader: Derivation[ConfigReader[Config]]): ConfigReader.Result[Config] = {
-    files.map(parseFile)
-      .map {
-        case Left(failures) if failures.toList.exists(_.isInstanceOf[CannotReadFile]) && !failOnReadError =>
-          Right(ConfigFactory.empty())
-        case conf => conf
-      }
-      .foldLeft[ConfigReader.Result[TypesafeConfig]](Right(ConfigFactory.empty())) {
-        case (c1, c2) => ConfigReader.Result.zipWith(c1, c2)(_.withFallback(_))
-      }
-      .right.flatMap { conf => loadConfig[Config](conf.resolve, namespace) }
+    files.map(ConfigSource.file).foldLeft(ConfigSource.empty)(filesReduceFunc(failOnReadError)).at(namespace).load[Config]
```

Comment:
This changes the behavior:
1. It fails if the first file doesn't exist.
1. It doesn't load `reference.conf`.

If we had an `optional` combinator, we could write:
```scala
files.map(ConfigSource.file(_).optional)
  .foldLeft(ConfigSource.defaultReference)(_.withFallback(_))
  .at(namespace)
  .load[Config]
```